src/backend/opencl/kernel/morph.cl

```diff
@@ -33,12 +33,14 @@ void morph(__global T *              out,
            KParam                  iInfo,
            __constant const T *   d_filt,
            __local T *          localMem,
-           int nBBS0, int nBBS1)
+           int nBBS0, int nBBS1, int windLen)
 {
-    const int halo   = windLen/2;
-    const int padding= 2*halo;
-    const int shrdLen = get_local_size(0) + padding + 1;
-    const int shrdLen1= get_local_size(1) + padding;
+    windLen = (SeLength>0 ? SeLength : windLen);
```

Comment:
ternary operator causes a lot of problems with some compilers. Can you make this an if statement?